tests/run_selective.py

```
@@ -57,18 +65,18 @@
 		suite = get_test_suite(args.group, args.category)
 		runner = unittest.TextTestRunner(verbosity=2)
 		result = runner.run(suite)
-		if (sys.stdout.isatty()):
-			print(f"{_bar} END {_bar}", file=sys.stdout)
+		logger.info(f"{_bar} END {_bar}")  # skipcq PYL-W1203 - test code ok
 		del _bar  # skipcq - cleanup any object leaks early
 		sys.exit(not result.wasSuccessful())
 	except ValueError as e:
-		if (sys.stderr.isatty()):
-			print(f"Error: {e}", file=sys.stderr)
+		logger.exception(f"Error: {e}")
```

Ruff (0.8.2)

Logging statement uses f-string

(G004)